addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -252,11 +253,19 @@ class HrWorkEntryType(models.Model):
     active = fields.Boolean(
         'Active', default=True,
         help="If the active field is set to false, it will allow you to hide the work entry type without removing it.")
+    country_id = fields.Many2one('res.country', string="Country")
 
     _sql_constraints = [
         ('unique_work_entry_code', 'UNIQUE(code)', 'The same code cannot be associated to multiple work entry types.'),
     ]
```

Comment:
I feel like this constraint should become a python constraint to avoid codes available for the same country
```py
@api.constrains('code', 'country_id')
def _check_code_unicity(self):
    similar_work_entry_types = self.search([
        ('code', 'in', self.mapped('code')),
        ('country_id', 'in', self.country_id.ids + [False]),
    ])
    for work_entry_type in self:
        if similar_work_entry_types.filtered_domain([
            ('code', '=', work_entry_type.code),
            ('country_id', 'in', self.country_id.id + [False]),
        ]):
            raise UserError(_("The same code cannot be associated to multiple work entry types."))

Reply:
do you think this is a good idea I mean we are using code for calculation of work entries I think we should use unique code for every work entry type.

Reply:
Ok i will do the changes